ethcore/builtin/src/lib.rs

```diff
@@ -198,111 +214,151 @@ impl ModexpPricer {
 ///
 /// Unless `is_active` is true,
 pub struct Builtin {
-	pricer: Box<dyn Pricer>,
-	native: Box<dyn Implementation>,
-	activate_at: u64,
+	pricer: BTreeMap<u64, Pricing>,
+	native: EthereumBuiltin,
 }
 
 impl Builtin {
 	/// Simple forwarder for cost.
+	///
+	/// Return the cost of the most recently activated pricer at the current block number.
+	///
+	/// If no pricer is actived `zero` is returned
+	///
+	/// If multiple `activation_at` has the same block number the last one is used
+	/// which follows the BTreeMap semantics
+	#[inline]
 	pub fn cost(&self, input: &[u8], at: u64) -> U256 {
-		self.pricer.cost(input, at)
+		if let Some((_, pricer)) = self.pricer.range(0..=at).last() {
+			pricer.cost(input)
+		} else {
+			U256::zero()
+		}
 	}
 
 	/// Simple forwarder for execute.
+	#[inline]
 	pub fn execute(&self, input: &[u8], output: &mut BytesRef) -> Result<(), &'static str> {
 		self.native.execute(input, output)
 	}
 
 	/// Whether the builtin is activated at the given block number.
+	#[inline]
```

Comment:
Just curious: did you measure that these `#[inline]`s matter or is it hunch-based optimization?

Reply:
I think those were already inlined thus I didn't see any measurable difference IIRC, do want the annotations removed?

Reply:
No, it's fine, I was just curious.